### ISSUE TITLE
Issue with Sleep

### ISSUE THREAD

**Issue by bararchy** [manastech/crystal#722]
Version 0.7.2ת Ubuntu 14.04 Using the repo

Code:
```crystal
def self.process_stop(log)
      log.info("Got stop request for RSAccess service, trying stop") if log
      pid = File.read((Dir.glob("/tmp/Reverse.Access.*.Service.exe.lock"))[0]).to_i rescue 0
      return 1 if pid == 0
      Process.kill(9, pid) rescue nil
      log.info("daemon sent kill -9 command")
      sleep 0.5
      log.info("daemon sent kill -9 command")
      begin
        Process.getpgid( pid )
        log.error("Failed to Stop RSAccess main process") if log
      return 1
    rescue
        log.error("RSAccess process has stopped successfully") if log
        File.delete((Dir.glob("/tmp/Reverse.Access.*.Service.exe.lock"))[0]) rescue nil
        return 0
    end
  end
```

Expected result:

```
daemon sent kill -9 command
daemon sent kill -9 command
```

What happens:

```
daemon sent kill -9 command
sleeps forever
```

**Comment by ysbaddaden** [comment]
Would you happen to run this inside a thread? Then your thread crashed, and exceptions aren't reported until you join it: any IO call and sleep related calls are currently broken inside threads since the libevent integration.

**Comment by asterite** [comment]
@barachy Please, always provide snippets that we can copy, paste, run and verify that they work incorrectly. We can't just guess how you are using that `process_stop` method. I also guess it has something to do with threads.

**Comment by bararchy** [comment]
Sorry ,

It was inside a Thread , I will try to always supply a working code to test .

Will this issue happens also when Fibers are used ?

Sent with AquaMail for Android
http://www.aqua-mail.com

On June 1, 2015 6:53:01 PM Ary Borenszweig <notifications@github.com> wrote:

@barachy Please, always provide snippets that we can copy, paste, run and verify that they work incorrectly. We can't just guess how you are using that process_stop method. I also guess it has something to do with threads.

-

**Comment by ysbaddaden** [comment]
TL;DR don't use threads or even fibers until further notice, and only `spawn { ... }` coroutines. They will run on a single thread for now, but will eventually be dispatched over any threads (I hope it will come fast).

I understand it's a bit saddening to lose threads for CPU-bound apps, but the libevent loop is faster than threads when it comes to IO and timers, and you can spawn *a lot* of coroutines.

**Comment by bararchy** [comment]
@ysbaddaden  Thanks for the explanation.

I'll work on changing my Thread.new to spawn do :)
